packages/design-picker/src/components/style-variation-badges/badge.tsx

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+import { getPreviewStylesFromVariation } from './utils';
+import type { StyleVariation } from '../../types';
+import './style.scss';
+
+interface BadgeProps {
+	variation?: StyleVariation;
+}
+
+const Badge: React.FC< BadgeProps > = ( { variation } ) => {
+	const styles = useMemo(
+		() => variation && getPreviewStylesFromVariation( variation ),
+		[ variation ]
+	);
+
+	if ( ! styles ) {
+		return null;
+	}
+
+	return (
+		<div className="style-variation__badge-wrapper">
+			<span
+				style={ {
+					backgroundColor: styles.color.background,
+					color: styles.color.foreground,
```

Comment:
Just want to check: Do styles always have color properties?

Reply:
[Yes](https://opengrok.a8c.com/source/xref/wpcom/wp-content/rest-api-plugins/endpoints/starter-designs/starter-designs.php?r=7abe900e#226)

Reply:
[Yes](https://github.com/Automattic/wp-calypso/pull/67066/files#diff-6d72b2f09c8547b51089bc672d6ee3a20fb1e29efd03712d0a60811197f00051R49) (at least on Calypso)